Source/DafnyStandardLibraries/src/Std/Parsers/Core/ParsersBuilders.dfy

```diff
@@ -0,0 +1,501 @@
+/*******************************************************************************
+ *  Copyright by the contributors to the Dafny Project
+ *  SPDX-License-Identifier: MIT 
+ *******************************************************************************/
+
+// Simple DSL to build parsers to avoid too much parenthesis nesting
```

Comment:
You should probably either document the missing methods or not bother with the summary (I'd lean towards the latter, you've documented them individually very well and the examples give a good summary of the common ones already)

Reply:
I lean towards the latter as well. Here is the rewritten summary

```
// A domain-specific language (DSL) for building parsers that uses minimal syntax
// (including single-letter combinators like M or C) to visually separate the parser logic
// (constants, strings, and custom parser calls) from the underlying combinator
// machinery. This makes complex parser definitions more readable by reducing
// syntactic noise.
```